module/modmanager/modmaninterface/interface.go

```diff
@@ -0,0 +1,27 @@
+// Package modmaninterface abstracts the manager interface to avoid an import cycle/loop.
+package modmaninterface
+
+import (
+	"context"
+
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/resource"
+)
+
+// ModuleManager abstracts the module manager interface.
+type ModuleManager interface {
+	AddModule(ctx context.Context, cfg config.Module) error
+
+	AddComponent(ctx context.Context, cfg config.Component, deps []string) (interface{}, error)
```

Comment:
how close can we get to having this be just for resources from an API perspective. they're so close

Reply:
I know! It's so frustrating because the only difference (now that services have dependencies) is that config.Component is a different type than config.Service. I need to do the rebase/merge to bring that in, and then see what else would be needed.

Reply:
AddComponent/Service is combined now, using that hack-ish thing we discussed today.